crates/testing/src/script.rs

```diff
@@ -93,11 +145,14 @@ pub async fn run_test_script<
     TYPES,
     S: TaskState<Event = Arc<HotShotEvent<TYPES>>> + Send + 'static,
 >(
-    mut script: TestScript<TYPES, S>,
+    // mut script: TestScript<TYPES, S>,
+    script: Vec<TestScriptStage<TYPES, S>>,
     state: S,
 ) where
     TYPES: NodeType,
 {
+    let mut script: TestScript<TYPES, S> = script.into();
```

Comment:
I really think we should not do this here. The actual execution of the script should be deterministic and shouldn't shuffle its inputs, because we lose the ability to insist on an order.

But I have no objection to adding a wrapper function that randomizes the script before call this and using that everywhere (i.e. have `run_test_script_random` and `run_test_script` or `run_test_script` and `run_test_script_deterministic`).

Reply:
Yeah this right now is just a hack, I plan to make this much less terrible. I just wanted to get it working before I edited every single file.